MSAL/src/MSALPublicClientApplication.m

```diff
@@ -744,15 +765,36 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
     params.loginHint = loginHint;
     params.extraAuthorizeURLQueryParameters = extraQueryParameters;
     params.accountIdentifier = account.lookupAccountIdentifier;
-    params.validateAuthority = _validateAuthority;
-    params.extraURLQueryParameters = _sliceParameters;
-    params.tokenExpirationBuffer = _expirationBuffer;
-    params.extendedLifetimeEnabled = _extendedLifetimeEnabled;
-    params.clientCapabilities = _clientCapabilities;
+    
+    params.extraURLQueryParameters = _configuration.extraQueryParameters.extraURLQueryParameters;
 
+    NSMutableDictionary *extraAuthorizeURLQueryParameters = [_configuration.extraQueryParameters.extraAuthorizeURLQueryParameters mutableCopy];
+    [extraAuthorizeURLQueryParameters addEntriesFromDictionary:extraQueryParameters];
+    params.extraAuthorizeURLQueryParameters = extraAuthorizeURLQueryParameters;
+    params.extraTokenRequestParameters = _configuration.extraQueryParameters.extraTokenURLParameters;
+    
+    params.tokenExpirationBuffer = _configuration.tokenExpirationBuffer;
+    params.extendedLifetimeEnabled = _configuration.extendedLifetimeEnabled;
+    params.clientCapabilities = _configuration.clientApplicationCapabilities;
+
+    params.validateAuthority = _configuration.validateAuthority
+        || [MSIDAADAuthority isAuthorityFormatValid:_configuration.authority.url context:nil error:nil];
```

Comment:
So, even if authority validation is off, we still validate it? Not sure it's necessary.

Reply:
For AAD, no? If not can totally remove :) No biggie here.